packages/ui/src/containers/PhoneForm/use-sms-register.ts

```diff
@@ -0,0 +1,55 @@
+import { SignInIdentifier } from '@logto/schemas';
+import { useState, useMemo, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { sendRegisterSmsPasscode } from '@/apis/register';
+import type { ErrorHandlers } from '@/hooks/use-api';
+import useApi from '@/hooks/use-api';
+import { UserFlow } from '@/types';
+
+const useSmsRegister = () => {
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const navigate = useNavigate();
+
+  const errorHandlers: ErrorHandlers = useMemo(
+    () => ({
+      'guard.invalid_input': () => {
+        setErrorMessage('invalid_phone');
+      },
+    }),
+    [setErrorMessage]
+  );
+
+  const clearErrorMessage = useCallback(() => {
+    setErrorMessage('');
+  }, []);
```

Comment:
Can we use the react-hook-form error state instead of maintaining it ourselves?

Reply:
that's an old topic can't recall.  react-hook-form is not used in main flow. Will migrate at our SSR project.